bestofjs/bestofjs
# Add missing "Featured projects" feature to beta version

## Goal

Add missing "Featured projects" to the beta version (following the feedback we got here #178 )

Some context about this feature: the goal was to highlight a subset of important projects, from the top page, showing the daily trends about projects pick randomly to try to entertain users.

This PR:

- Adds a section "Featured projects" in the top page that shows 5 featured projects randomly (visible on desktop only)
- Adds `/features` page to let users browse through all featured projects

I did a cleanup about the data, adding a specific status value `featured` in this commit https://github.com/bestofjs/bestofjs-backend/commit/21ed1533e9048d3284d1a57dffcfbe9f96684a9d . This status is currently assigned to about 200 projects.

## How to test

- In the home page, check the behavior of the "Featured" sections, clicking on "previous" and "next" buttons to cycle through the 200 features projects
- Click on "View more" button to go to `/featured` page and check the pagination and the search order option. By default the most recent additions in _Best of JS_ show up first.

## Screenshot, 1st iteration

#### home page

![image](https://github.com/bestofjs/bestofjs/assets/5546996/128f444e-067c-4d20-bf47-fd1e29985667)

#### `/featured` page

![image](https://github.com/bestofjs/bestofjs/assets/5546996/ff3bf998-0a85-48f8-af81-0656dff0ee1d)
